src/Application/Digital/Commands/ShareEntry.cs

```diff
@@ -0,0 +1,178 @@
+        public bool CanDownload { get; init; }
+        public bool CanChangePermission { get; init; }
+    }
+    
+    public class CommandHandler : IRequestHandler<Command, EntryPermissionDto>
+    {
+        private readonly IApplicationDbContext _context;
+        private readonly IMapper _mapper;
+        private readonly IDateTimeProvider _dateTimeProvider;
+
+        public CommandHandler(IApplicationDbContext applicationDbContext, IMapper mapper, IDateTimeProvider dateTimeProvider)
+        {
+            _context = applicationDbContext;
+            _mapper = mapper;
+            _dateTimeProvider = dateTimeProvider;
+        }
+        
+        public async Task<EntryPermissionDto> Handle(Command request, CancellationToken cancellationToken)
+        {
+            var entry = await _context.Entries
+                .Include(x => x.Uploader)
+                .Include(x => x.Owner)
+                .FirstOrDefaultAsync(x => x.Id == request.EntryId, cancellationToken);
+            
+            if (entry is null)
+            {
+                throw new KeyNotFoundException("Entry does not exist.");
+            }
+
+            var hasModifyPermission = _context.EntryPermissions.Any(x =>
+                x.EntryId == request.EntryId
+                && x.EmployeeId == request.CurrentUser.Id
+                && x.AllowedOperations.Contains(EntryOperation.ChangePermission.ToString()));
+            
+            if (entry.Owner!.Id != request.CurrentUser.Id && !hasModifyPermission)
+            {
+                throw new UnauthorizedAccessException("You are not allow to change permission of this entry.");
+            }
+            
+            var user = await _context.Users.FirstOrDefaultAsync(x => x.Id == request.UserId, cancellationToken);
+
+            if (user is null)
+            {
+                throw new KeyNotFoundException("User does not exist.");
+            }
+
+            if (user.Id == entry.OwnerId)
+            {
+                throw new ConflictException("You can not modify owner's permission.");
+            }
+
+            if (request.ExpiryDate < _dateTimeProvider.DateTimeNow)
+            {
+                throw new ConflictException("Expiry date cannot be in the past.");
+            }
+
+            var allowOperations = GenerateAllowOperations(request, entry.IsDirectory);
+            
+            await GrantPermission(entry, user, allowOperations, request.ExpiryDate, cancellationToken);
+
+            if (entry.IsDirectory)
+            {
+                // Update permissions for child Entries
+                var path = entry.Path.Equals("/") ? (entry.Path + entry.Name) : (entry.Path + "/" + entry.Name);
+                var childEntries = _context.Entries
+                    .Include(x => x.Uploader)
+                    .Include(x => x.Owner)
+                    .Where(x => x.Id != entry.Id 
+                                && x.Path.StartsWith(path)
```

Comment:
this is not realistic, what if I'm updating /a and I have another path called /ab ? Something better could be 
```
var path = entry.Path.Equals("/") ? entry.Path + entry.Name : $"{entry.Path}/{entry.Name}";
var pattern = $"{path}/%";
var childEntries = _context.Entries.Where(x => 
    x.Path.Trim().ToLower().Equals(path)
    || EF.Functions.Like(x.Path, pattern));

```

Reply:
True. But without `ToLower()` since `/aA/a` shouldn't be child of `/aa`

Reply:
should the user use sensitive or insensitive input?

Reply:
fixed at https://github.com/ProFile-org/ProFile-BE/pull/336/commits/13d51e4e8a15ef33a9d64d0abaf9265b9914ee6a